### ISSUE TITLE
PhysicalDeviceFeatures2Builder::push_next()?

### ISSUE THREAD

**Issue by cheako** [ash-rs/ash#325]
I'm trying to get_physical_device_features2() for PhysicalDeviceDescriptorIndexingFeatures.
Edit: maybe VkPhysicalDeviceVulkan12Features?  Same issue.

**Comment by filnet** [comment]
relates to https://github.com/MaikKlein/ash/pull/305

**Comment by cheako** [comment]
Thinking about it, I don't know what cause PhysicalDeviceFeatures2 to ever be used as a branch.  It can be, but it's main usage would be for get_physical_device_features2().  Afterwards DeviceCreateInfo.p_enabled_features == PhysicalDeviceFeatures2.features, so there is currently no reason to have PhysicalDeviceFeatures2 be a branch of DeviceCreateInfo.

**Comment by aclysma** [comment]
I just hit this issue too. I think this is a functional workaround?

```rust
        let mut descriptor_indexing_features = ash::vk::PhysicalDeviceDescriptorIndexingFeaturesEXT::default();
        let mut features_v2 = vk::PhysicalDeviceFeatures2::default();

        unsafe {
            features_v2.p_next = &mut descriptor_indexing_features as *mut _ as *mut std::ffi::c_void;
            instance.get_physical_device_features2(device, &mut features_v2)
        };
```

**Comment by MarijnS95** [comment]
This function is now available since #305 (in Ash release 0.33) where we finally cleaned up how `structextends` is handled. Any struct that is mentioned one or more times in a `structextends` gets a `trait ExtendsXXX` and `fn push_next()`, and that trait is implemented for the struct mentioning it in `structextends`. And it is removed from all structs that are never extended according to `vk.xml`, getting rid of unnecessary bloat.
